server/src/main/java/org/elasticsearch/health/GetHealthAction.java

```diff
@@ -142,7 +167,13 @@ public TransportAction(
 
         @Override
         protected void doExecute(Task task, Request request, ActionListener<Response> listener) {
-            listener.onResponse(new Response(clusterService.getClusterName(), healthService.getHealth()));
+            listener.onResponse(
+                new Response(
+                    clusterService.getClusterName(),
+                    healthService.getHealth(request.componentName, request.indicatorName, request.computeDetails),
+                    request.componentName == null && request.indicatorName == null
```

Comment:
I'll remove the `&& request.indicatorName == null` because it's not needed -- one of the two Request constructors sets both to `null`, and the other asserts that `componentName` is not null.